src/cli/driver.ts

```diff
@@ -555,16 +555,27 @@ export class Driver {
       let suppressionUsed = false;
       warnings.forEach((projectWarnings, projectName) => {
         const filteredWarnings = projectWarnings.filter((warning) => {
-          const lc = warning.loc.interval.getLineAndColumn() as {
-            lineNum: number;
-            colNum: number;
-          };
+          if (!warning.loc.file) {
+            return true;
+          }
+          // Get the base directory from the warning file
+          const warningDir = path.dirname(warning.loc.file);
+          // Convert warning path to absolute
+          const warningFile = warning.loc.file;
+          // Resolve relative to the warning directory's parent
+          const suppressionFile = path.isAbsolute(suppression.file)
+            ? suppression.file
+            : path.resolve(path.dirname(warningDir), suppression.file);
```

Comment:
Why do we call `path.dirname` for `warningDir` twice?

Reply:
`path.resolve` resolves paths from right-most to left-most. I'm not sure why it might work here.